src/reaper_osara.cpp

```diff
@@ -2424,6 +2424,20 @@ void cmdMoveToPrevItemKeepSel(Command* command) {
 	}
 }
 
+void cmdPropertiesFocus(Command* command) {
+	switch (fakeFocus) {
+		case FOCUS_ITEM:
+		case FOCUS_TAKEMARKER:
+			Main_OnCommand(40009, 0); // Item properties: Show media item/take properties
+			break;
+		case FOCUS_AUTOMATIONITEM:
+			Main_OnCommand(42090, 0); // Envelope: Automation item properties...
+			break;
+		default:
+			break;
```

Comment:
While this does make sense for a new command, because we're looking to replace the existing item properties command, I worry that users will be confused when they press shift+f2 when focus is on the ruler (just as an example) and nothing happens. On the other hand, it's probably weird if they move to an auto item, move on the ruler, press properties and end up with media item properties.

My suggestion is that if we don't know how to handle the focus (this default block), we should make a decision based on the shouldMoveToAutoItem variable. If true, use auto item properties. If false, use media item properties. shouldMoveToAutoItem gets set to false whenever you move to a track and set to true whenever you move to an envelope or envelope point.

That makes me wonder whether we should even bother looking at fakeFocus at all at this stage or just make the decision purely based on that variable. Obviously, that might have to change in future if we start adding more properties things to this command.

Reply:
This makes sense to me. Changed as suggested.